federation/service/node.go

```diff
@@ -0,0 +1,203 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"math/rand"
+	"strconv"
+	"time"
+
+	"github.com/cortezaproject/corteza-server/federation/types"
+	"github.com/cortezaproject/corteza-server/federation/util"
+	"github.com/cortezaproject/corteza-server/pkg/actionlog"
+	"github.com/cortezaproject/corteza-server/pkg/auth"
+	"github.com/cortezaproject/corteza-server/pkg/id"
+	"github.com/cortezaproject/corteza-server/store"
+	"github.com/cortezaproject/corteza-server/system/service"
+	stypes "github.com/cortezaproject/corteza-server/system/types"
+	"github.com/davecgh/go-spew/spew"
+)
+
+var (
+	// This is temporary for local testing
+	tmpNodeStore = make(types.NodeSet, 0)
+
+	ErrorNodeNotFound   = errors.New("pair: node not found")
+	ErrorInvalidNodeURI = errors.New("pair: invalid node uri provided")
+)
+
+type (
+	node struct {
+		ctx context.Context
+
+		actionlog actionlog.Recorder
+
+		store        store.Storable
+		sysUser      service.UserService
+		tokenEncoder auth.TokenEncoder
+	}
+
+	NodeService interface {
+		With(ctx context.Context) NodeService
+
+		Create(n *types.Node, sharedDomain string) (*types.Node, error)
+		Pair(nodeID uint64) error
+		HandshakeInit(sourceNodeID, nodeIDB uint64, nodeURI, token string) error
+		HandshakeConfirm(nodeID uint64) error
+		HandshakeComplete(nodeID uint64, token string) error
+	}
+)
+
+func Node(ctx context.Context) NodeService {
+	rand.Seed(time.Now().UnixNano())
+	return (&node{
+		store: DefaultNgStore,
+
+		actionlog:    DefaultActionlog,
+		tokenEncoder: auth.DefaultJwtHandler,
+	}).With(ctx)
+}
+
+func (svc node) With(ctx context.Context) NodeService {
```

Comment:
Remove this and follow the standard go-way with context.Context as 1st arg in the functions (where needed)
This (with(ctx) pattern is OBSOLETE, DANGEROUS and MUST NOT be used when implementing new services.